# First Steps with Python and Jupyter 



![](https://i.imgur.com/gvSnw4A.png)




This tutorial covers the following topics:

* Performing arithmetic operations using Python
* Solving multi-step problems using variables
* Evaluating conditions using Python
* Combining conditions with logical operators
* Adding text styles using Markdown

## Performing Arithmetic Operations using Python

Let's begin by using Python as a calculator. You can write and execute Python using a code cell within Jupyter. 

> **Working with cells**: To create a new cell within Jupyter, you can select "Insert > Insert Cell Below" from the menu bar or just press the "+" button on the toolbar. You can also use the keyboard shortcut `Esc+B` to create a new cell. Once a cell is created, click on it to select it. You can then change the cell type to code or markdown (text) using the "Cell > Cell Type" menu option. You can also use the keyboard shortcuts `Esc+Y` and `Esc+M`. Double-click a cell to edit the content within the cell. To apply your changes and run a cell, use the "Cell > Run Cells" menu option or click the "Run" button on the toolbar or just use the keyboard shortcut `Shift+Enter`. You can see a full list of keyboard shortcuts using the "Help > Keyboard Shortcuts" menu option.

Run the code cells below to perform calculations and view their result. Try changing the numbers and run the modified cells again to see updated results. Can you guess what the `//`, `%`, and `**` operators are used for?

In [6]:
print('Hello: Good Mornning')

Hello: Good Mornning


In [1]:
2 + 3 + 9

14

In [3]:
99 - 73

26

In [5]:
23.54 * -1432

-33709.28

In [7]:
100 / 8

12.5

In [9]:
100 // 8

12

In [11]:
100 % 8

4

In [13]:
5 ** 3

125

As you might expect, operators like `/` and `*` take precedence over other operators like `+` and `-` as per mathematical conventions. You can use parentheses, i.e. `(` and `)`, to specify the order in which operations are performed.

In [15]:
2 + 5 * 17 - 3 / 4 ** 3

86.953125

In [17]:
((2 + 5) * (17 - 3)) / (4 ** 3)

1.53125

In [12]:
6%2

0

In [18]:
100 / 7

14.285714285714286

In [19]:
100//7

14

In [20]:
100 % 7

2

In [74]:
4 % 3

1

In [75]:
14 * 7

98

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |


Try solving some simple problems from this page:
https://www.math-only-math.com/worksheet-on-word-problems-on-four-operations.html . 

You can use the empty cells below and add more cells if required.

In [7]:
5 / 3

1.6666666666666667

In [8]:
5 // 3

1

In [9]:
5 % 3

2

In [76]:
Var = 8

In [77]:
Var 

8

In [78]:
Var = 18

In [79]:
Var

18

## Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [19]:
500 * (.2 * 1.25)

125.0

Thus, the grocery store makes a total profit of $125. While this is a reasonable way to solve a problem, it's not entirely clear by looking at the code cell what the numbers represent. We can give names to each of the numbers by creating Python *variables*.

> **Variables**: While working with a programming language such as Python, information is stored in *variables*. You can think of variables as containers for storing data. The data stored within a variable is called its *value*.

In [53]:
cost_of_ice_bag = 1.25

In [55]:
cost_of_ice_bag

1.25

In [57]:
print(cost_of_ice_bag)

1.25


In [59]:
profit_margin = .2

In [61]:
number_of_bags = 500

The variables `cost_of_ice_bag`, `profit_margin`, and `number_of_bags` now contain the information provided in the word problem. We can check the value of a variable by typing its name into a cell. We can combine variables using arithmetic operations to create other variables.

> **Code completion**: While typing the name of an existing variable in a code cell within Jupyter, just type the first few characters and press the `Tab` key to autocomplete the variable's name. Try typing `pro` in a code cell below and press `Tab` to autocomplete to `profit_margin`.

In [50]:
profit_margin

0.2

In [63]:
profit_per_bag = cost_of_ice_bag * profit_margin

In [65]:
profit_per_bag 

0.25

In [67]:
number_of_bags

500

In [69]:
total_profit = number_of_bags * profit_per_bag

In [71]:
total_profit

125.0

In [73]:
print(total_profit)

125.0


If you try to view the value of a variable that has not been *defined*, i.e., given a value using the assignment statement `variable_name = value`, Python shows an error.

In [79]:
gross_profits 

NameError: name 'gross_profits' is not defined

In [66]:
gross_profit

59

In [68]:
net_profit = total_profit/5

In [70]:
net_profit

25.0

Storing and manipulating data using appropriately named variables is a great way to explain what your code does.

Let's display the result of the word problem using a friendly message. We can do this using the `print` *function*.

> **Functions**: A function is a reusable set of instructions. It takes one or more inputs, performs certain operations, and often returns an output. Python provides many in-built functions like `print` and also allows us to define our own functions.

In [76]:
'The grocery store makes a total profit of $'

'The grocery store makes a total profit of $'

In [77]:
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


> **`print`**: The `print` function is used to display information. It takes one or more inputs, which can be text (within quotes, e.g., `"this is some text"`), numbers, variables, mathematical expressions, etc. We'll learn more about variables & functions in the next tutorial.

Creating a code cell for each variable or mathematical operation can get tedious. Fortunately, Jupyter allows you to write multiple lines of code within a single code cell. The result of the last line of code within the cell is displayed as the output. 

Let's rewrite the solution to our word problem within a single cell.

In [85]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
profit_per_bag

total_profit = number_of_bags * profit_per_bag
total_profit

125.0

In [87]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
print('Profit Per bag: ', profit_per_bag)

total_profit = number_of_bags * profit_per_bag
print("Total Profit: ", total_profit)

# Display the result
print("The grocery store makes a total profit of $", total_profit)

Profit Per bag:  0.25
Total Profit:  125.0
The grocery store makes a total profit of $ 125.0


Note that we're using the `#` character to add *comments* within our code. 

> **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines. 

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `"""`. Here are some examples of code comments:

In [89]:
my_favorite_number = 1 # an inline comment

In [91]:
# This comment gets its own line
# Thsidndnd
# ththtjtj
# thtjtjjtjt'
# thhtjt
my_least_favorite_number = 3

In [ ]:
print('Hey Solomon!')

In [93]:
"""This is a multi-line comment.
Write as little or as much as you'd like.

Comments are really helpful for people reading
your code, but try to keep them short & to-the-point.

Also, if you use good variable names, then your code is
often self explanatory, and you may not even need comments!
"""
a_neutral_number = 5
a_neutral_number

5



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [95]:
56 == 56

True

In [97]:
'56' == 56

False

In [99]:
56 * 5

280

In [101]:
'56' * 5

'5656565656'

In [107]:
SOLOMON = 'SOLOMON'

In [109]:
SOLOMON == 'SOLOMON'

True

In [111]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [115]:
# Equality check - True
my_favorite_number == 1

True

In [117]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [123]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [125]:
a_neutral_number

3

In [121]:
# Not equal check - False
a_neutral_number != 3

False

In [127]:
# Greater than check - True
my_least_favorite_number > a_neutral_number

True

In [129]:
# Greater than check - False
my_favorite_number > my_least_favorite_number

False

In [106]:
# Less than check - True
my_favorite_number < 10

True

In [131]:
# Less than check - False
my_least_favorite_number < my_favorite_number

False

In [133]:
my_favorite_number

1

In [110]:
# Greater than or equal check - True
my_favorite_number >= 1

True

In [135]:
# Greater than or equal check - False
my_favorite_number >= 3

False

In [137]:
# Less than or equal check - True
3 + 6 <= 9

True

In [75]:
# Less than or equal check - False
my_favorite_number + a_neutral_number <= 3

False

Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

In [76]:
cost_of_ice_bag = 1.25

is_ice_bag_expensive = cost_of_ice_bag >= 10

In [77]:
is_ice_bag_expensive

False

In [78]:
print("Is the ice bag expensive?", is_ice_bag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |


In [ ]:
& | ~

In [139]:
5 + 7 > 2 - 18

True

In [145]:
12 // 3 == 4

True

In [147]:
5 + 7 > 2 - 18 and 12 // 3 == 4

True

In [149]:
my_favorite_number

1

In [153]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [155]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [81]:
my_favorite_number > 0 and my_favorite_number >= 3

False

In [157]:
True and False

False

In [159]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |



In [161]:
a_neutral_number

3

In [163]:
a_neutral_number != 3 

False

In [173]:
my_favorite_number > 0

True

In [175]:
a_neutral_number != 3 or my_favorite_number > 0

True

In [177]:
a_neutral_number != 3 or my_favorite_number > 0

True

In [65]:
my_favorite_number

1

In [66]:
my_favorite_number < 0 or True

True

In [185]:
True or False

True

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [190]:
not True

False

In [192]:
not False

True

In [85]:
print(not True)
print(not False)

False
True


In [48]:
a_neutral_number == 3

True

In [49]:
not a_neutral_number == 3

False

In [51]:
not my_favorite_number < 0

True

In [52]:
not False

True

In [53]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [54]:
False or not False

True

In [58]:
(my_favorite_number < 0 and True)

False

In [55]:
(2 > 3 and 4 <= 5) or not (my_favorite_number < 0 and True)

True

In [70]:
2 > 3 and 4 <= 5

False

In [71]:
(my_favorite_number < 0 and True)

False

In [72]:
not (my_favorite_number < 0 and True)

True

In [121]:
my_favorite_number < 0 and True

False

In [122]:
(2 > 3 and 4 <= 5)

False

In [123]:
(my_favorite_number < 0 and True)

False

In [124]:
(2 > 3 and 4 <= 5)

False

In [125]:
not False

True

In [126]:
False or not False

True

In [127]:
not (True and 0 < 1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [128]:
not True

False

In [129]:
0 < 1

True

In [130]:
False and True

False

In [131]:
not True and 0 < 1 or False and True

False

In [132]:
not True

False

In [133]:
False and 0 < 1 or False and True

False

In [134]:
False and 0 < 1

False

Experiment with arithmetic, conditional and logical operators in Python using the interactive nature of Jupyter notebooks. We will learn more about variables and functions in future tutorials.

## Solution for Exercise



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [135]:
plane_cost=5000
total_people_signed_up=29
ticket_cost=200

In [136]:
amount_received_by_company=total_people_signed_up*ticket_cost

In [137]:
total_profit=amount_received_by_company-plane_cost

In [138]:
total_profit

800

The travel company is in a profit of $800!